File: backend/app/models/diagnosis.py
from pydantic import BaseModel
from typing import Optional, List

class DiagnoseRequest(BaseModel):
    complaint: str
    error_code: Optional[str] = None

class ErrorCodeMatch(BaseModel):
    code: str
    meaning: str
    severity: str
    possible_causes: Optional[str] = None
    general_action: Optional[str] = None

class SuggestedIssue(BaseModel):
    issue_id: int
    issue_name: str
    category: str
    confidence: float
    severity: str
    diy_difficulty: str
    estimated_time: int
    parts_needed: Optional[str] = None
    tools_required: Optional[str] = None
    symptoms: Optional[str] = None
    possible_causes: Optional[str] = None

class DiagnoseResponse(BaseModel):
    error_code_match: Optional[ErrorCodeMatch] = None
    suggested_issues: List[SuggestedIssue]
    recommendation: str  # "try_diy" or "schedule_technician"

File: mobile/src/api/diagnosis.ts
const BACKEND_URL = 'http://192.168.1.11:8000';

export interface CommonIssue {
  issue_id: number;
  issue_name: string;
  category: string;
  confidence: number;
  severity: string;
  diy_difficulty: string;
  estimated_time: number;
  parts_needed?: string;
  tools_required?: string;
  symptoms?: string;
  possible_causes?: string;
  affected_parts_3d?: string;
}

export interface ErrorCodeInfo {
  code: string;
  meaning: string;
  severity: string;
  possible_causes?: string;
  general_action?: string;
}

export interface DiagnosisResult {
  issue: CommonIssue;
  confidence: number;
  errorCode?: ErrorCodeInfo;
}

export async function fetchDiagnosis(
  complaint: string,
  errorCode?: string
): Promise<DiagnosisResult[]> {
  try {
    const response = await fetch(`${BACKEND_URL}/api/diagnose`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({
        complaint,
        error_code: errorCode ?? null,
      }),
    });

    if (!response.ok) throw new Error(`HTTP ${response.status}`);

    const data = await response.json();

    const issues = data.suggested_issues as CommonIssue[];
    const maxConf = issues[0]?.confidence ?? 1;

    return issues.map((issue) => ({
      issue,
      confidence: Math.round((issue.confidence / maxConf) * 87),
      errorCode: data.error_code_match ?? undefined,
    }));

  } catch (e) {
    console.error('Diagnosis API error:', e);
    return [];
  }
}